src/test/java/core/basesyntax/service/strategy/PurchaseHandlerTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service.strategy;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.exceptions.ProductQuantityException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseHandlerTest {
+    private Storage storage;
+    private BalanceHandler handler;
+    private PurchaseHandler purchaseHandler;
+    private final String fruit = "apple";
+    private final int quantity = 20;
+    private final int subtractQuantity = 5;
+
+    @BeforeEach
+    void setUp() {
+        storage = new Storage();
+        handler = new BalanceHandler(storage);
+        purchaseHandler = new PurchaseHandler(storage);
```

Comment:
You don't need to create before each method new services if you don't change them